commands/createfixpullrequests.go

```diff
@@ -0,0 +1,196 @@
+	"os/exec"
+	"strings"
+)
+
+func CreateFixPullRequests(c *clitool.Context) error {
+	// Get params and VCS client
+	params, client, err := utils.GetParamsAndClient()
+	if err != nil {
+		return err
+	}
+	// Send usage report
+	usageReportSent := make(chan error)
+	go utils.ReportUsage(c.Command.Name, &params.Server, usageReportSent)
+
+	// Do scan commit
+	scanResults, err := scanCommit(params)
+	if err != nil {
+		return err
+	}
+
+	// Fix and create PRs
+	err = fixImpactedPackagesAndCreatePRs(params, client, scanResults)
+	if err != nil {
+		return err
+	}
+
+	// Wait for usage report
+	<-usageReportSent
+	return err
+}
+
+// Audit the dependencies of the current commit.
+func scanCommit(params *utils.FrogbotParams) ([]services.ScanResponse, error) {
+	// Audit commit code
+	xrayScanParams := createXrayScanParams(params.Watches, params.Project)
+	scanResults, err := auditSource(xrayScanParams, params)
+	if err != nil {
+		return nil, err
+	}
+	clientLog.Info("Xray scan completed")
+	return scanResults, nil
+}
+
+// Audit the dependencies of the current branch.
+func fixImpactedPackagesAndCreatePRs(params *utils.FrogbotParams, client vcsclient.VcsClient, scanResults []services.ScanResponse) error {
+	fixVersionsMap, err := createFixVersionsMap(scanResults)
+	if err != nil {
+		return err
+	}
+	clientLog.Info("Found", len(fixVersionsMap), "impacted packages with fix versions")
+
+	gitManager, err := utils.NewGitManager(".", "origin")
+	if err != nil {
+		return err
+	}
+	for impactedPackage, fixVersionInfo := range fixVersionsMap {
+		clientLog.Info("Fixing", impactedPackage, "with", fixVersionInfo.fixVersion)
+		err = fixSinglePackageAndCreatePR(impactedPackage, *fixVersionInfo, params, client, gitManager)
+		if err != nil {
+			clientLog.Error("failed while trying to fix and create PR for:", impactedPackage, "with version:", fixVersionInfo.fixVersion, "with error:", err.Error())
+		}
+	}
+	return nil
+}
+
+// Create vulnerabilities rows. The rows should contain only the new issues added by this PR
+func createFixVersionsMap(scanResults []services.ScanResponse) (map[string]*FixVersionInfo, error) {
+	fixVersionsMap := map[string]*FixVersionInfo{}
+	for _, scanResult := range scanResults {
+		if len(scanResult.Violations) > 0 {
+			// todo!
+		} else if len(scanResult.Vulnerabilities) > 0 {
+			vulnerabilities, err := xrayutils.PrepareVulnerabilities(scanResult.Vulnerabilities, false, false)
+			if err != nil {
+				return nil, err
+			}
+			for _, vulnerability := range vulnerabilities {
+				if vulnerability.FixedVersions != nil && len(vulnerability.FixedVersions) > 0 {
+					fixVersion := parseVersionChangeString(vulnerability.FixedVersions[0])
+					fixVersionInfo, exists := fixVersionsMap[vulnerability.ImpactedPackageName]
+					if exists {
+						fixVersionInfo.UpdateFixVersion(fixVersion)
+					} else {
+						fixVersionsMap[vulnerability.ImpactedPackageName] = NewFixVersionInfo(fixVersion, vulnerability.ImpactedPackageType)
```

Comment:
Let's explain here why we are doing this, in case the fixed version doesn't exist.